pkg/scheduler/algorithm/priorities/requested_to_capacity_ratio.go

```diff
@@ -84,35 +86,64 @@ func NewFunctionShape(points []FunctionShapePoint) (FunctionShape, error) {
 	return pointsCopy, nil
 }
 
+// NewResources creates instance of Resources in a safe way performing all
+// necessary sanity checks.
+func NewResources(resources Resources) (Resources, error) {
+	n := len(resources)
+
+	if n == 0 {
+		resources = Resources{v1.ResourceMemory: 1, v1.ResourceCPU: 1}
+	}
```

Comment:
As the old implementation of ResourceToCapacityRatio did not have a resourceToWeightMap for backward compatibility I was setting the default value.

Reply:
you are still returning some default value here, that I think we should not. Just return an error when resourceToWeightMap is empty (the correct place for initialization is what you are doing in line 110).

Reply:
I am using the defaultValue set above because the old function did not require these params to be set by the user. I  have made the resourceToWeight Parameter optional for backward compatibility.